src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/action/RollupActionIT.kt

```diff
@@ -0,0 +1,228 @@
+                pageSize = 100,
+                dimensions = listOf(
+                        DateHistogram(sourceField = "tpep_pickup_datetime", fixedInterval = "1h"),
+                        Terms("RatecodeID", "RatecodeID"),
+                        Terms("PULocationID", "PULocationID")
+                ),
+                metrics = listOf(
+                        RollupMetrics(sourceField = "passenger_count", targetField = "passenger_count", metrics = listOf(Sum(), Min(), Max(),
+                                ValueCount(), Average())),
+                        RollupMetrics(sourceField = "total_amount", targetField = "total_amount", metrics = listOf(Max(), Min()))
+                )
+        )
+        val rollupId = rollup.toRollup(indexName).id
+        val actionConfig = RollupActionConfig(rollup, 0)
+        val states = listOf(
+                State("rollup", listOf(actionConfig), listOf())
+        )
+        val sourceIndexMappingString = "\"properties\": {\"tpep_pickup_datetime\": { \"type\": \"date\" }, \"RatecodeID\": { \"type\": " +
+                "\"keyword\" }, \"PULocationID\": { \"type\": \"keyword\" }, \"passenger_count\": { \"type\": \"integer\" }, \"total_amount\": " +
+                "{ \"type\": \"double\" }}"
+        val policy = Policy(
+                id = policyID,
+                description = "$testIndexName description",
+                schemaVersion = 1L,
+                lastUpdatedTime = Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                errorNotification = randomErrorNotification(),
+                defaultState = states[0].name,
+                states = states
+        )
+        createPolicy(policy, policyID)
+        createIndex(indexName, policyID, mapping = sourceIndexMappingString)
+
+        val managedIndexConfig = getExistingManagedIndexConfig(indexName)
+
+        // Change the start time so the job will initialize the policy
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+        waitFor { assertEquals(policyID, getExplainManagedIndexMetaData(indexName).policyID) }
+
+        // Change the start time so we attempt to create rollup step will execute
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+        waitFor {
+            assertEquals(
+                    AttemptCreateRollupJobStep.getSuccessMessage(rollupId, indexName),
+                    getExplainManagedIndexMetaData(indexName).info?.get("message")
+            )
+        }
+
+        Thread.sleep(60000)
+
+        // Change the start time so wait for rollup step will execute
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+        waitFor {
+            assertEquals(
+                    WaitForRollupCompletionStep.getJobCompletionMessage(rollupId),
+                    getExplainManagedIndexMetaData(indexName).info?.get("message")
+            )
+        }
+
+        waitFor {
+            val rollupJob = getRollup(rollupId = rollupId)
```

Comment:
can move this waitFor into the above waitFor

Reply:
```
waitFor {
    val rollupJob = getRollup(rollupId = rollupId)
    assertNotNull("Rollup job doesn't have metadata set", rollupJob.metadataID)
    val rollupMetadata = getRollupMetadata(rollupJob.metadataID!!)
    assertEquals("Rollup is not finished", RollupMetadata.Status.FINISHED, rollupMetadata.status)
    assertEquals(
            WaitForRollupCompletionStep.getJobCompletionMessage(rollupId),
            getExplainManagedIndexMetaData(indexName).info?.get("message")
    )
}
```

Reply:
updated